src/server/rest/v1/service/TenantService.ts

```diff
@@ -326,6 +329,9 @@ export default class TenantService {
   }
 
   private static async updateSettingsWithComponents(tenant: Partial<Tenant>, req: Request): Promise<void> {
+    // Check if OICP component is activated or deactivated and create/activate/deactivate virtual user (and Badges) accordingly
+    await this.checkOICPComponent(tenant);
```

Comment:
Okay, but how is this method triggered if not here?

If the Master component settings change I should set the virtual user inactive etc.

I can move checkOICPComponent(tenant) to OICPUtils. But where should it be called?

Reply:
In that PR https://github.com/LucasBrazi06/ev-server/pull/1874, you have a method called createInitialSettingsForTenant() that have been created for that purpose. 
Ask @LucasBrazi06 to merge it (it's approved) and add there the code or the call to the method that will init OICP settings.

Reply:
#1874 is my PR.

Do you have an idea where to trigger the function to set the OICP virtual user active/inactive in case that the OICP components gets deactivated?

Reply:
> #1874 is my PR.
> 
> Do you have an idea where to trigger the function to set the OICP virtual user active/inactive in case that the OICP components gets deactivated?

In the REST endpoint that receive the components enablement or not per tenant.